env-check.sh

```diff
@@ -16,22 +19,31 @@ func_check_python_pkg(){
         if python3 -c "import $1" &> /dev/null; then
             return
         else
-            out_str=$out_str"\nPlease install \e[31m python3-$1 \e[0m package"
+            out_str="$out_str""\nPlease install \e[31m python3-$1 \e[0m package"
             check_res=1
         fi
     else
         return
     fi
 }
 
+func_check_file(){
+    if [ -e "$1" ]; then
+        return
+    fi
+    out_str="$out_str""\nOptional: Enable dynamic debug logs on \e[31m $1 \e[0m file\n\tFor example,\n\toptions snd_sof dyndbg=+p\n\toptions snd_sof_pci dyndbg=+p"
+    check_res=1
+}
+
 check_res=0
 echo -ne "Check for the package:\t\t"
 func_check_pkg expect
 func_check_pkg aplay
 func_check_pkg python3
 func_check_python_pkg graphviz
+func_check_file "$DYNDBG"
 [[ $check_res -eq 0 ]] && echo "pass" || \
-    echo -e "\e[31mWarning\e[0m\nSolution:"$out_str
+    echo -e "\e[31mWarning\e[0m\nSolution:""$out_str"
```

Comment:
Thanks for fixing these quoting issues but:
- in a different and earlier commit
- use `printf '\e[31mWarning\e[0m\nSolution: %s\n' "$out_str"`
https://www.google.com/search?q=echo+printf

Reply:
Thank you for reviewing this. If I just replace printf with echo -e, special characters are just came out. And line change also messed up.
```
Solution:\nPlease install \e[31m expect \e[0m package\nOptional: Enable dynamic debug logs on \e[31m /etc/modprobe.d/sof-dyndbg.conf \e[0m file\n\tFor example,\n\toptions snd_sof dyndbg=+p\n\toptions snd_sof_pci dyndbg=+p
Check for tools folder:         Warning
Solution:\n
\tMissing execution permission of script/binary in tools folder\n
\tWarning: you need to make sure the current user has execution permssion\n
\tPlease use the following command to give execution permission:\n
\t\e[31mcd .\n
\tchmod a+x tools/*\e[0m
Checking for case folder:               pass
Checking the permission:                Warning
Solution:\n
\tMissing permission to access sound card\n
\t\tPlease use the following command to add current user to the audio group:\n
\t\e[31msed -i '/^audio:/s:$:,fredoh:g' /etc/group\e[0m
Checking the config setup:              pass
```

Came up with "%b" trick. This is the output from my build machine.
```
$ ./env-check.sh
Check for the package:          Warning
Solution:
Please install  expect  package
Optional: Enable dynamic debug logs on  /etc/modprobe.d/sof-dyndbg.conf  file
        For example,
        options snd_sof dyndbg=+p
        options snd_sof_pci dyndbg=+p
Check for tools folder:         Warning
Solution:
        Missing execution permission of script/binary in tools folder
        Warning: you need to make sure the current user has execution permssion
        Please use the following command to give execution permission:
        cd .
        chmod a+x tools/*
Checking for case folder:               pass
Checking the permission:                Warning
Solution:
        Missing permission to access sound card
                Please use the following command to add current user to the audio group:
        sed -i '/^audio:/s:$:,fredoh:g' /etc/group
Checking the config setup:              pass
```

Reply:
Sorry I missed there are \e escape codes in the `$out_str` argument too.

`%b` works, otherwise this should also work and is easier to read?
```shell
printf '\e[31mWarning\e[0m\nSolution:'"$out_str"'\n' # no argument, everything in the format string
# OK as long as we control what goes into `$out_str`
```

Reply:
> Thanks for fixing these quoting issues but: in a different and earlier commit

Please?